src/main/java/seedu/address/model/person/PersonMatchesPredicate.java

```diff
@@ -0,0 +1,108 @@
+package seedu.address.model.person;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
+ */
+public class PersonMatchesPredicate implements Predicate<Person> {
+
+    private List<String> namesList;
+    private List<String> moduleList;
+
+    private boolean hasNamesList;
+    private boolean hasModuleList;
+
+    public PersonMatchesPredicate() {
+        namesList = new ArrayList<>();
+        moduleList = new ArrayList<>();
+        hasNamesList = false;
+        hasModuleList = false;
+    }
+
+    @Override
+    public boolean test(Person person) {
```

Comment:
Possible to rename ```test``` to fit Java coding standards.

Reply:
Thanks for the comment. The `PersonMatchesPredicate` class implements the `Predicate` interface so it needs to override the `test` method.